keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -450,157 +488,101 @@ about permitting exports from `kube-system` and `default`. As a best practice,
-unreachable to the supercluster controller, each `EndpointSlice` is associated
-with a lease representing connectivity with its source cluster. The supercluster
-service controller is responsible for periodically renewing the lease so long as
-the connection with the source cluster is confirmed alive. A separate
-controller, that may run inside each cluster, is responsible for watching each
-lease and removing all remaining `EndpointSlices` associated with a cluster when
-that cluster’s lease expires.
-
-#### Service Types
-
-- `ClusterIP`: This is the the straightforward case most of the proposal
-  assumes. Each `EndpointSlice` associated with the exported service is combined
-  with slices from other clusters to make up the supercluster service. They will
-  be imported to the cluster behind the supercluster IP.
-
-```
-<<[UNRESOLVED re:stateful sets]>>
-  Today's headless services likely don't want a VIP and may not function
-  properly behind one. It probably doesn't make sense to export a current
-  headless service to the supercluster, it would work, but likely not the way
-  you want.
-<<[/UNRESOLVED]>>
-```
-- `NodePort` and `LoadBalancer`: These create `ClusterIP` services that would
-  sync as expected. For example If you export a `NodePort` service, the
-  resulting cross-cluster service will still be a supercluster IP type. You
-  could use node ports to access the cluster-local service in the source
-  cluster, but not in any other cluster, and it would only route to local
-  endpoints.
-- `ExternalName`: It doesn't make sense to export an `ExternalName` service.
-  They can't be merged with other exports, and it seems like it would only
-  complicate deployments by even attempting to stretch them across clusters.
-  Instead, regular `ExternalName` type `Services` should be created in each
-  cluster individually.
-
-### Consumption of EndpointSlice
-
-To consume a supercluster service, users will use the domain name associated
-with their `ServiceExport`. When the mcsd-controller sees a `ServiceExport`, a
-`ServiceImport` will be introduced, which can be largely ignored by the user. 
-
-An `ServiceImport` is a service that may have endpoints in other clusters.
+A `ServiceImport` is a service that may have endpoints in other clusters.
 This includes 3 scenarios:
-1. This service is running entirely in different cluster(s)
-1. This service has endpoints in other cluster(s) and in this cluster
-1. This service is running entirely in this cluster, but is exported to other cluster(s) as well
+1. This service is running entirely in different cluster(s).
+2. This service has endpoints in other cluster(s) and in this cluster.
+3. This service is running entirely in this cluster, but is exported to other
+   cluster(s) as well.
+
+A multi-cluster service will be imported only by clusters in which the service's
+namespace exists. All clusters containing the service's namespace will import
```

Comment:
My understanding of the conversation was that we would allow a controller to decide whether or not to automatically create the namespace, but that if the namespace exists we will always import the service. Reason being that a "do-not-import" label seems to go against the concept of namespace sameness and could lead to quite a bit more complexity if left wide open. If we see demand we can certainly revisit, but let's start with the simplest case.

Reply:
I feel like this is over-constraining, but I guess it's easier to loosen the spec later..